### PR TITLE
Adding cggi-tfhe-rs-bool pass + Add tests + And_packed

### PR DESCRIPTION
**Pull request by WoutLegiest**
- Adding packed version of the And operation  to the tfhe-rs-bool emitter
- Adding two tests where an add operation is implemented using boolean operations, for cggi to tfhe-rs-bool and tfhe-rs-bool code emitting
- Organising of the tfhe-rs e2e tests;

### CODE REVIEW

File: lib/Conversion/CGGIToTfheRustBool/CGGIToTfheRustBool.cpp
Comment (j2kun) [inline lib/Conversion/CGGIToTfheRustBool/CGGIToTfheRustBool.cpp:239]:
```suggestion
```

File: lib/Conversion/CGGIToTfheRustBool/CGGIToTfheRustBool.cpp
Comment (j2kun) [inline lib/Conversion/CGGIToTfheRustBool/CGGIToTfheRustBool.cpp:291]: If you remove the associated patterns, I think you can also remove this line.

File: tests/tfhe_rust_bool/ops.mlir
Comment (j2kun) [inline tests/tfhe_rust_bool/ops.mlir:31]: If all you're doing is checking syntax, you can just make the arguments to and_packed the same as the function arguments and skip the setup.

File: lib/Conversion/CGGIToTfheRustBool/CGGIToTfheRustBool.cpp
Comment (WoutLegiest) [reply lib/Conversion/CGGIToTfheRustBool/CGGIToTfheRustBool.cpp:77]: I tried and tested it and the conversision of lwe to tfhe-rs-bool stopped working. Kept it in

File: include/Dialect/TfheRustBool/IR/TfheRustBoolOps.td
Comment (WoutLegiest) [reply include/Dialect/TfheRustBool/IR/TfheRustBoolOps.td:47]: I tried to (quickly) implement it but stumbled upon some errors. If it is oke, I try to incorporate it in coming PRs? 
Going to extend the packed operations for the fpga's

Comment (j2kun) [reply include/Dialect/TfheRustBool/IR/TfheRustBoolOps.td:47]: Sounds good!

File: lib/Conversion/CGGIToTfheRustBool/CGGIToTfheRustBool.cpp
Comment (j2kun) [reply lib/Conversion/CGGIToTfheRustBool/CGGIToTfheRustBool.cpp:77]: I'll take a look after it's merged in then.

**Review by j2kun — APPROVED** [review]

**Comment by j2kun** [comment]
You will need to squash your commits, but otherwise  LGTM!

**Pull request merged**
